src/classes/BusinessLogicServices/CarrierService.php

```diff
@@ -160,6 +166,10 @@ public function delete(ShippingMethod $shippingMethod)
      */
     public function addBackupShippingMethod(ShippingMethod $shippingMethod)
     {
+        /** @var ShippingMethodService $shippingMethodService */
+        $shippingMethodService = ServiceRegister::getService(ShippingMethodService::CLASS_NAME);
+        $shippingMethod = $shippingMethodService->getShippingMethod($shippingMethod->getId());
```

Comment:
Because the passed shipping method entity does not have all the necessary data on itself (this was discovered during testing). Method setCarrierData could not get the delivery time from the ShippingMethod entity because that property was not set. Therefore, in order to get all the required data, the shipping method had to be fetched again from the database.

Reply:
In such cases, pass only the ID and explain why.

Reply:
The issue was with the passing object. This has been fixed in the Core.